web/src/components/vis/GroupPredictionTile.vue

```diff
@@ -0,0 +1,356 @@
+    };
+  },
+  computed: {
+    // Column names (i.e. metabolites) available for selection,
+    // filtered based on current metabolite source
+    columns() {
+      if (this.metaboliteSource === 'all') {
+        return this.dataset.column.data.filter((column) => column.column_type === 'measurement')
+          .map((column) => column.column_header);
+      }
+      if (this.metaboliteSource === 'selected') {
+        return this.dataset.selectedColumns;
+      }
+      const { metabolites } = this.pcaData;
+      if (!metabolites) {
+        return [];
+      }
+      return metabolites.filter((m, i) => (this.pcaData.factor[i] === this.metaboliteSource));
+    },
+    groups() {
+      return this.dataset.groupLevels.map((level) => level.name);
+    },
+    rocData() {
+      if (!this.plot.data || !this.group1 || !this.group2 || this.metabolites.length === 0) {
+        // if any data is missing, don't draw the ROC curve
+        return { sensitivities: [], specificities: [], auc: 0 };
+      }
+      return {
+        sensitivities: this.plot.data.sensitivities,
+        specificities: this.plot.data.specificities,
+        auc: this.plot.data.auc[0],
+      };
+    },
+  },
+  watch: {
+    // get new factor analysis when threshold changes
+    threshold() {
+      this.getFactors();
+      this.metaboliteSource = 'all';
+    },
+
+    // Clear metabolites when metabolite source is changed to 'All'
+    // If the source is changed to 'Selected' or one of the PC factors,
+    // populate the list with the filtered metabolite and graph them.
+    metaboliteSource(newSource) {
+      this.metabolites = newSource === 'all' ? [] : this.columns;
+      this.changePlotArgs({ columns: JSON.stringify(this.metabolites) });
+    },
+
+    // These two watchers prevent the same group from being selected twice.
+    group1(newGroup, oldGroup) {
+      this.changePlotArgs({ group1: newGroup });
+      if (this.group2 === newGroup) {
+        this.group2 = oldGroup;
+        this.changePlotArgs({ group2: oldGroup });
+      }
+    },
+    group2(newGroup, oldGroup) {
+      this.changePlotArgs({ group2: newGroup });
+      if (this.group1 === newGroup) {
+        this.group1 = oldGroup;
+        this.changePlotArgs({ group1: oldGroup });
+      }
+    },
+  },
+  mounted() {
+    // Perform an initial factor analysis when component is mounted
+    this.getFactors();
+  },
+  methods: {
+    // Called when a metabolite is removed from the ROC analysis
+    // as a result of the user clicking the red 'X' next to its
+    // name in the 'Metabolites' autoselect
+    removeMetabolite(metabolite) {
+      this.metabolites.splice(this.metabolites.indexOf(metabolite), 1);
+      this.changePlotArgs({ columns: JSON.stringify(this.metabolites) });
+    },
+    async getFactors() {
+      if (this.metaboliteSourceOptions.length > 2) {
+        this.metaboliteSourceOptions.splice(2, this.metaboliteSourceOptions.length - 2);
+      }
```

Comment:
Rather than having to do this stitching to keep the default options in the metabolite source list, why not use a computed variable? You're already storing `pcaData`, so you can simply make `metaboliteSourceOptions` computed with something like this:
```
metaboliteSourceOptions() {
  let options = [... default options ...];
  if (this.pcaData && this.pcaData.factors) {
    options.concat(this.pcaData.factor.map((factor) => { value: factor, text: `PC${factor}` }));
  }
}
```

Reply:
Addressed in https://github.com/girder/viime/pull/587/commits/90fe6832a3dacef3ca4f9c92bba216192163d129.